### PR TITLE
CI develop

### PR DESCRIPTION
**Pull request by OlegChub**
Add API tests
Add CI steps to Teamcity
Add Dockerfile
Add slack notifier and RP launch result reporting

### CODE REVIEW

**Comment by OlegChub** [comment]
RP
Slack
TC build run
TC build steps
TC trigger

File: src/main/java/api/service/SlackService.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    private static SlackConfig config = ConfigFactory.create(SlackConfig.class, System.getProperties());
+
+    public static void postNotification(String message) {
+        JSONObject request = new JSONObject();
+        request.put("text", message);
+        String postUrl = String.format("https://hooks.slack.com/services/%s", config.token());
+        given()
+                .header("Content-type", "application/json")
+                .body(request.toJSONString()).
+                when()
```
Comment (Fentezi3) [inline src/main/java/api/service/SlackService.java]: move dot from 17th line to 18th

File: src/test/java/APITests.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+        JSONObject request = new JSONObject();
+        request.put("name", "Lera");
+        request.put("job", "logistics");
+
+        given()
+                .header("Content-type", "application/json")
+                .contentType(ContentType.JSON)
+                .accept(ContentType.JSON)
+                .body(request.toJSONString()).
+                when()
```
Comment (Fentezi3) [inline src/test/java/APITests.java]: move dot from 46th line to 17th

File: src/test/java/APITests.java [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+                .post("https://reqres.in/api/users").
+                then()
```
Comment (Fentezi3) [inline src/test/java/APITests.java]: move dot from 48th line to 49th

File: src/test/java/APITests.java [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+    }
+
+    @Test
+    public void testPATCH() {
+        JSONObject request = new JSONObject();
+        request.put("job", "logistics");
+
+        given()
+                .body(request.toJSONString()).
+                when()
```
Comment (Fentezi3) [inline src/test/java/APITests.java]: move dot from 74th line to 75th

File: src/test/java/APITests.java [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+                .patch("https://reqres.in/api/users/2").
+                then()
```
Comment (Fentezi3) [inline src/test/java/APITests.java]: move dot from 76th line to 77th

File: src/test/java/APITests.java [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+                .statusCode(200)
+                .log().all();
+    }
+
+    @Test
+    public void testDELETE() {
+        when()
+                .delete("https://reqres.in/api/users/2").
+                then()
```
Comment (Fentezi3) [inline src/test/java/APITests.java]: move dot from 85th line to 86th

**Review by Fentezi3 — APPROVED** [review]

**Review by dsalonka — APPROVED** [review]

File: src/main/java/api/service/SlackService.java (hunk 1 above)
Comment (OlegChub) [reply src/main/java/api/service/SlackService.java]: I didn't pay the attention to these tests at all, cause this is not the goal of the HW. But nevertheless, fixed.

File: src/test/java/APITests.java (hunk 2 above)
Comment (OlegChub) [reply src/test/java/APITests.java]: I didn't pay the attention to these tests at all, cause this is not the goal of the HW. But nevertheless, fixed.

File: src/test/java/APITests.java (hunk 3 above)
Comment (OlegChub) [reply src/test/java/APITests.java]: I didn't pay the attention to these tests at all, cause this is not the goal of the HW. But nevertheless, fixed.

File: src/test/java/APITests.java (hunk 4 above)
Comment (OlegChub) [reply src/test/java/APITests.java]: I didn't pay the attention to these tests at all, cause this is not the goal of the HW. But nevertheless, fixed.

File: src/test/java/APITests.java (hunk 5 above)
Comment (OlegChub) [reply src/test/java/APITests.java]: I didn't pay the attention to these tests at all, cause this is not the goal of the HW. But nevertheless, fixed.

File: src/test/java/APITests.java (hunk 6 above)
Comment (OlegChub) [reply src/test/java/APITests.java]: I didn't pay the attention to these tests at all, cause this is not the goal of the HW. But nevertheless, fixed.

**Review by dKatechev — APPROVED** [review]

**Pull request merged**